### ISSUE TITLE
wrong stack_size when on another thread

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7527]
**Original bug ID:** 7527
**Reporter:** domsj
**Assigned to:** @mshinwell
**Status:** assigned (set by @mshinwell on 2017-05-04T15:37:10Z)
**Resolution:** open
**Priority:** normal
**Severity:** minor
**Version:** 4.03.0
**Category:** standard library

## Bug description

```ocaml
let () =
  let print_stack_size () =
    let open Gc in
    Printf.printf "stack_size=%i\n" (quick_stat ()).stack_size;
  in
  print_stack_size ();
  Thread.create print_stack_size () |> Thread.join
```

this snippet prints:
```
stack_size=26
stack_size=1035826253803286485
```

**Comment by vicuna** [comment]
**Comment author:** @mshinwell

This should be fixed by:

https://github.com/ocaml/ocaml/pull/1062

which I shall get back to as soon as I can.

**Comment by xavierleroy** [comment]
Fixed by #1062 and #8670 .
